scripts/lmdb_matrix.py

```diff
@@ -0,0 +1,164 @@
+import lmdb
+import json
+import click
+from dataclasses import dataclass
+
+from contextlib import contextmanager
+import numpy as np
+
+
+@dataclass
+class GenotypeMatrix:
+    """Store the actual Genotype Matrix"""
+    matrix: np.ndarray
+    transpose: np.ndarray
+    metadata: dict
+
+
+@contextmanager
+def save_excursion(stream):
+    """Context manager to restore stream position after use."""
+    position = stream.tell()
+    try:
+        yield
+    finally:
+        stream.seek(position)
+
+
+def count_lines(stream) -> int:
+    """Count the number of lines in the stream from the current
+    position."""
+    count = 0
+    with save_excursion(stream):
+        while stream.readline().strip():
+            count += 1
+    return count
+
+
+def create_database(db_path: str) -> lmdb.Environment:
+    """Create or open an LMDB environment."""
+    return lmdb.open(db_path, map_size=100 * 1024 * 1024, create=True)
+
+
+def genotype_db_put(db: lmdb.Environment, genotype: GenotypeMatrix) -> bool:
+    metadata = json.dumps(genotype.metadata).encode("utf-8")
+    with db.begin(write=True) as txn:
+        txn.put(b"matrix", genotype.matrix.tobytes())
+        txn.put(b"transpose", genotype.transpose.tobytes())
```

Comment:
Question: Why do you need to store the  transposed matrix?

Reply:
I've removed this in my updated work.  You are right, we don't need this for now; since that adds un-necessary overhead for a really huge matrix.